packages/plugin-ext/src/main/node/plugin-deployer-impl.ts

```diff
@@ -83,8 +88,16 @@ export class PluginDeployerImpl implements PluginDeployer {
         // init resolvers
         await this.initResolvers();
 
-        // check THEIA_DEFAULT_PLUGINS or THEIA_PLUGINS env var
-        const defaultPluginsValue = process.env.THEIA_DEFAULT_PLUGINS || undefined;
+        // check the `--plugins` CLI option and THEIA_PLUGINS env var
+        let defaultPluginsValue = this.cliContribution.localDir();
+        if (!defaultPluginsValue) {
+            // check THEIA_DEFAULT_PLUGINS env var. To be backward compatible after GH-4150.
+            defaultPluginsValue = process.env.THEIA_DEFAULT_PLUGINS;
+            if (defaultPluginsValue) {
+                // tslint:disable-next-line:max-line-length
+                this.logger.warn(`Setting the location of the plugins via the 'THEIA_DEFAULT_PLUGINS' environment variable is deprecated. Use the --${PluginCliContribution.PLUGINS}=${LocalDirectoryPluginDeployerResolver.LOCAL_DIR}:path/to/plugins CLI option instead.`);
```

Comment:
Hello @kittaakos didn't see your previous ping

We shouldn't say that it's deprecated as it's not. (The env variable is still something that can be used)
Some instances of theia are not started with the cli and may never (like EclipseChe/theia)

Please just do the merge of all three options in array of all these data:
cliContrib.localDir() + defaultPluginsValue + pluginsValue

into `pluginsList`

Reply:
@benoitf, I made the changes based on your feedback. I hope this is what you had in mind. Thanks!

Reply:
yes that's perfect @kittaakos 👍